pkg/cmd/roachtest/test_runner.go

```diff
@@ -1723,6 +1736,79 @@ func (r *testRunner) teardownTest(
 	return nil
 }
 
+// inspectArtifacts inspects node logs and attempts to write helpful triage
+// information to the test log and testRunner.githubMessage
+// This method is best effort and should not fail a test.
+func (r *testRunner) inspectArtifacts(
+	ctx context.Context, t *testImpl, testLogger *logger.Logger, timedOut bool,
+) error {
+
+	if timedOut {
+		testLogger.Printf("Test timed out. Do not attempt to inspect artifacts.")
+		return nil
+	}
+
+	var errMonitor registry.FatalMonitorError
+	if !failuresMatchingError(t.failures(), &errMonitor) {
+		return nil
+	}
+
+	testLogger.Printf("Test has failed due to fatal error observed by the monitor. " +
+		"Attempting to gather node fatal level logs for triage.")
+	out, err := gatherFatalNodeLogs(t, testLogger)
+	if err != nil {
+		return err
+	}
+
+	msg := fmt.Sprintf("CockroachDB contains Fatal level logs. Up to the first 10 "+
+		"will be shown here. See node logs in artifacts for more details.\n%s", out)
+	testLogger.PrintfCtx(ctx, msg)
+	t.appendGithubMessage(msg)
+
+	return nil
+}
+
+// gatherFatalNodeLogs attempts to gather fatal level node logs to help with
+// triage
+func gatherFatalNodeLogs(t *testImpl, testLogger *logger.Logger) (string, error) {
+	logPattern := `^F[0-9]{6}`
+	filePattern := "logs/*unredacted/cockroach*.log"
+	// *unredacted captures patterns for single node and multi-node clusters
+	// e.g. unredacted, 1.unredacted
+	joinedFilePath := filepath.Join(t.ArtifactsDir(), filePattern)
+	targetFiles, err := filepath.Glob(joinedFilePath)
+	if err != nil {
+		return "", err
+	} else if len(targetFiles) == 0 {
+		return "", errors.Newf("No matching log files found for log pattern: %s and file pattern: %s",
+			logPattern, filePattern)
+	}
+	args := append([]string{"-E", "-m", "10", logPattern}, targetFiles...)
+	command := "grep"
+	// Note: need to wrap the regex in quotes if you want to copy this command from the logs and
+	// use directly
+	t.L().Printf("Gathering fatal level logs with command: %s %s", command, strings.Join(args, " "))
+	cmd := exec.Command("grep", args...)
```

Comment:
The command should be executed (remotely) on all cluster nodes, not locally; the latter works only in local-mode.
It would also help (readability) to group the fatal message by node id.

Reply:
> Works with remote because we will always download the artifacts if there's a test failure (except for timeout).

Ah, I totally forgot. In fairness, that's a hidden assumption, so we should explicitly comment why this works.